src/utils/user.js

```diff
@@ -0,0 +1,53 @@
+export const addNewListToUser = (user, listName, newToken) => {
+	const [userToken, setUserToken] = user;
+	const tokenLists = JSON.parse(userToken);
+	setUserToken(JSON.stringify({ ...tokenLists, [listName]: newToken }));
+	return { ...tokenLists, [listName]: newToken };
+};
+
+export const setNewUserToken = (user, newToken, listName) => {
+	const [, setUserToken] = user;
+	if (listName) {
+		setUserToken(JSON.stringify({ [listName]: newToken }));
+		return { [listName]: newToken };
+	} else {
+		setUserToken(JSON.stringify({ newToken }));
+		return { newToken };
+	}
+};
+
+export const getUserListsArr = (userToken) =>
+	Object.entries(JSON.parse(userToken));
+
+export const setTokenFirstList = (setListToken, userLists) => {
+	if (Object.entries(userLists).length) {
+		setListToken(Object.entries(userLists)[0][1]);
+	} else setListToken(null);
+};
+
+export const updateName = (user, value, listToken) => {
+	const [userToken, setUserToken] = user;
+	const prevList = JSON.parse(userToken);
+
+	for (let name in prevList) {
+		if (prevList[name] === listToken) {
+			delete prevList[name];
+		}
+	}
+
+	setUserToken(JSON.stringify({ ...prevList, [value]: listToken }));
+	return { ...prevList, [value]: listToken };
+};
+
+export const getMatchingName = (userToken, matchToken) =>
+	getUserListsArr(userToken).find(([, token]) => token === matchToken)[0];
```

Comment:
Similar to above - the app generally works so wondering if `[,token]` was intentional

Reply:
yep! the first key in the array is unneeded so set it up this way.